src/nile/core/account.py

```diff
@@ -95,7 +109,7 @@ def declare(
     ):
         """Declare a contract through an Account contract."""
         if nonce is None:
-            nonce = get_nonce(self.address, self.network)
+            nonce = int(await get_nonce(self.address, self.network))
```

Comment:
doesn't `get_nonce` cast to int already?

Reply:
wondering if it makes sense to add a `get_nonce` function into this account module, that knows its own address and network

Reply:
I think it makes sense. It's probably better suited in a separate PR though